### PR TITLE
Lower the flow control defaults, remove psutil dependency

### LINKED ISSUE #5180: pubsub should have a default flow controller
re: https://github.com/GoogleCloudPlatform/google-cloud-python/issues/5058#issuecomment-378412550

PubSub should come configured with a default flow controller and some arbitrary default max messages (several other languages use 1000 as the number), with documentation around tuning it for higher-throughput workloads (which requires more memory).

### PR DESCRIPTION
**Pull request by jonparrott**
Closes #5180

### CODE REVIEW

**Comment by jadekler** [comment]
LGTM!

File: pubsub/setup.py [hunk 1]
```diff
@@ -31,7 +31,6 @@
 dependencies = [
     'google-api-core[grpc]<2.0.0dev,>=0.1.3',
     'grpc-google-iam-v1<0.12dev,>=0.11.1',
-    'psutil<6.0dev,>=5.2.2',
```
Comment (crwilcox) [inline pubsub/setup.py]: ty

**Comment by gylu** [comment]
I'm not sure what the accepted norm is, but just wondering... can changes like these be documented as "Breaking" in the future? Updating the PubSub library broke our system because these defaults changed and I was only looking for "Breaking Changes" instead of also "Implementation Changes"

**Comment by csoare7** [comment]
@gylu was wondering how did you manage to fix your PubSub library? This update also breaks our Cloud Storage, as we're getting `ModuleNotFoundError: No module named 'google'`

**Comment by gylu** [comment]
@csoare7 I rewrote the code. Hehe. I guess the lesson learned is to read all the release notes, and maybe update less often.

**Comment by tseaver** [comment]
@csoare7 You likely need to reinstall in a fresh environment, using current setuptools / pip.  E.g.:

```bash
$ python3.6 -m venv /path/to/new_venv
$ /path/to/new_venv/bin/pip install -U setuptools pip
...
Successfully installed pip-18.1 setuptools-40.4.3
$ /path/to/new_venv/bin/pip install google-cloud-pubsub google-cloud-storage
...
Successfully installed cachetools-2.1.0 certifi-2018.8.24 chardet-3.0.4 google-api-core-1.4.1 google-auth-1.5.1 google-cloud-core-0.28.1 google-cloud-pubsub-0.38.0 google-cloud-storage-1.13.0 google-resumable-media-0.3.1 googleapis-common-protos-1.5.3 grpc-google-iam-v1-0.11.4 grpcio-1.15.0 idna-2.7 protobuf-3.6.1 pyasn1-0.4.4 pyasn1-modules-0.2.2 pytz-2018.5 requests-2.19.1 rsa-4.0 six-1.11.0 urllib3-1.23
$ /path/to/new_venv/bin/python -c "import google; print(google.__path__)"
_NamespacePath(['/path/to/new_venv/lib/python3.6/site-packages/google'])
```

**Comment by tseaver** [comment]
@gylu Can you go into more detail about how your application broke?  The change in this PR should have been innocuous:  were you depending on having `psutil` installed via a transitive dependency?  Or was it the smaller defaults that broke you?

**Comment by csoare7** [comment]
@tseaver thanks for the comment, that's what I did locally and can confirm it works. However this breaks the test environment - we're running on Docker containers so every test build is a fresh environment.

**Pull request merged**